### DISCUSSION TITLE
Has anyone compiled darknet with CUDA 12 yet?

### DISCUSSION THREAD

**Discussion by AxelWS** [AlexeyAB/darknet#8836] [Q&A]
Hi all,
I tried to compile [build/darknet/yolo_cpp_dll.vcxproj](https://github.com/AlexeyAB/darknet/blob/d2c3788539bb73d3cfcdb7a1d5fa6b29ae1ec0e1/build/darknet/yolo_cpp_dll.vcxproj#L22) with CUDA 12. Currently it is set to CUDA 11. I had to convert it to Visual Studio 19 (Visual Studio 15 is no longer supported by CUDA 12). Then it did compile, but using the compiled DLL fails with an CUDA error in cuda_convert_f32_to_f16() when the network is created. Removing preprocessor define CUDNN_HALF changes the error to CUDNN_STATUS_NOT_SUPPORTED  in cudnn_convolutional_setup().
Has anyone else tried to compile with CUDA 12?

**Comment by mr-mainak** [comment]
When I'm performing make in CUDA=12.1 there is an error regarding ```-lcudnn``` flag. The linker is not available.

**Comment by stephanecharette** [comment]
Because you are trying to use an old unsupported repo.  The new darknet repo is here:  https://github.com/hank-ai/darknet#table-of-contents

**Comment by AxelWS** [comment]
@stephanecharette thank you for the hint, next time I will try hank-ai repo.
@saqib772 At first it looked like CUDA 11.7 on Ampere with AlexeyAB repo was working fine, but -map option seems to cause a crash, even with plenty of free GPU memory. Fortunately without -map training seems to run ok, so I fixed code to save weights more often and evaluate mAP and other metrics separately.

**Reply by AxelWS** [reply]
@saqib772 Have a look at commit b5ff7f4cd3803caf35d932f5a1d2529e445bbe3f "Save weights for each 1000 iterations if max_batches < 10000" and change detector.c as desired.

**Comment by stephanecharette** [comment]
> [...] but -map option seems to cause a crash, even with plenty of free GPU memory. Fortunately without -map training seems to run ok, so I fixed code to save weights more often and evaluate mAP and other metrics separately.

That is definitely the wrong fix.  You should not be training without `-map`.  Please use the new repo, where this bug (and many others!) is fixed.  https://github.com/hank-ai/darknet#table-of-contents
